<center> SEMANA  N° 2 : MATRICES </center>
<center> Víctor Mariano Villacorta Plasencia - 202013427</center>

In [1]:
import numpy as np

**1) Sean las matrices A y B siguientes:**

In [8]:
A = np.array([
    [20, 30, 50],
    [30, 20, 60],
    [30, 30, 32]
])
B = np.array([
    [5],
    [45],
    [10]
])

A.shape, B.shape

((3, 3), (3, 1))

**Las columnas de A representan los litros de
vino, ginebra y limonada que se han echado en
tres bidones para formar cocteles. Por su parte
la matriz B representa los precios por litro de
cada uno de los 3 líquidos.**

a) ¿Cuál es el significado real de la matriz
A.B?


In [3]:
A.dot(B)

array([[1950],
       [1650],
       [1820]])

RSPTA: El resultado nos muestra el precio total que costó las preparación de cada coctel 

b) Sea C la matriz 


In [12]:
C = np.array([3,4,5]).reshape(1,3)
C.shape

(1, 3)

 Hallar CAB y dar su significado real.  

In [15]:
ca = C.dot(A)
ca

array([[330, 320, 550]])

Hasta el momento tengo que para 3 vidones del 1er coctel, 4 vidones del 2do coctel y 5 vidones del 3er coctel obtengo 330 litros de vino, 320 litros de ginebra y 550 litros de limonada

In [17]:
cab = ca.dot(B)
cab

array([[21550]])

Conclusión final con 3 vidones del 1er coctel, 4 vidones del 2do coctel y 5 vidones del 3er coctel gasto 21550 soles

***2) Una firma de automóviles dispone de dos
plantas de fabricación una en España y otra en
Inglaterra, en los que fabrica dos modelos de
coches M1  y M2, de tres colores x, y, z. Su
capacidad de producción diaria en cada planta
está dada por las siguientes matrices (A para
España y B para Inglaterra).***

In [18]:
A = np.array([
    [300, 95],
    [250, 100],
    [200, 100]
])
B = np.array([
    [190, 90],
    [200, 100],
    [150, 80]
])

A.shape, B.shape

((3, 2), (3, 2))

a) Determinar la representación matricial de
la producción total por día. 


In [19]:
A + B

array([[490, 185],
       [450, 200],
       [350, 180]])

al sumar las matrices obtengo la producción total por modelo y color

b) Si se eleva la producción en España un 20%
y se disminuye en Inglaterra un 10% ¿qué
matriz representa la nueva producción
total? 


In [23]:
(120/100)*A + (90/100)*B

array([[531., 195.],
       [480., 210.],
       [375., 192.]])

esta seria la producción total con los porcentajes de afectación mencionados

***3) Un constructor hace una urbanización con tres
tipos de viviendas: S (sencillas), N (normales)
y L (lujo). Cada vivienda sencilla tiene 1
ventana grande, 7 medianas y 1 pequeña. Cada
vivienda normal tiene 2 ventanas grandes, 9
medianas y 2 pequeñas. Y cada vivienda de
lujo tiene 4 ventanas grandes, 10 medianas y 3
pequeñas.  
Cada ventana grande tiene 4 cristales y 8
bisagras, cada ventana mediana tiene dos
cristales y 4 bisagras; y cada ventana pequeña
tiene 1 cristal y 2 bisagras.***



a) Escribir una matriz que describa el número
y tamaño de ventanas en cada tipo de
vivienda; y otra matriz que exprese el
número de cristales y el número de
bisagras en cada tipo de ventana. 


In [24]:
VENTANA_CASA = np.array([
    [1, 2, 4],
    [7, 9, 10],
    [1, 2, 3]
])
CRISTALBISAGRA_VENTANA = np.array([
    [4, 2, 1],
    [8, 4, 2]
])

VENTANA_CASA.shape, CRISTALBISAGRA_VENTANA.shape

((3, 3), (2, 3))

b) Calcular una matriz que exprese el número
de cristales y de bisagras necesarias en cada
tipo de vivienda. 


In [25]:
CRISTALBISAGRA_CASA = CRISTALBISAGRA_VENTANA.dot(VENTANA_CASA)
CRISTALBISAGRA_CASA

array([[19, 28, 39],
       [38, 56, 78]])

- En casa simple hay 19 cristales y 38 bisagras
- En casa normal hay 28 cristales y 56 bisagras
- En casa lujo hay 39 cristales y 78 bisagras

***4) Una fábrica de muebles fabrica 3 modelos de
estanterías A, B y C, cada uno en dos tamaños
grande y pequeño. Produce diariamente 1000
estanterías grandes y 8000 estanterías
pequeñas del tipo A, 8000 grandes y 6000
pequeñas del tipo B y 4000 grandes y 6000
pequeñas del tipo C.  Cada estantería grande
lleva 16 tornillos y 6 soportes, y la pequeña
lleva 12 tornillos y 4 soportes.***


a) Representar esta información en dos
matrices. 

In [26]:
TAMANIO_ESTANTERIA = np.array([
    [1000, 8000, 4000],
    [8000, 6000, 6000]
])
TORNILLOSOPORTE_TAMANIO = np.array([
    [16, 12],
    [6, 4]
])

TAMANIO_ESTANTERIA.shape, TORNILLOSOPORTE_TAMANIO.shape

((2, 3), (2, 2))

b) Hallar una matriz que exprese la cantidad
de tornillos y soportes necesarios para la
producción diaria de cada uno de los seis
modelos-tamaño de estantería.


In [35]:
TORNILLOSOPORTE_ESTANTERIA = TORNILLOSOPORTE_TAMANIO.dot(TAMANIO_ESTANTERIA)
TORNILLOSOPORTE_ESTANTERIA

array([[112000, 200000, 136000],
       [ 38000,  72000,  48000]])

con esto tenemos que :
- Para el tipo A se necesitan 112000 tornillos y 38000 soportes
- Para el tipo B se necesitan 200000 tornillos y 72000 soportes
- Para el tipo C se necesitan 136000 tornillos y 48000 soportes

**5)  Una fábrica produce 3 artículos y tiene 4
clientes. El resumen mensual de ventas se
anota en una matriz, donde cada cliente
dispone de un vector fila cuyas componentes
indican las cantidades
adquiridas de cada
artículo. Sea E la
matriz de ventas de
enero:***


In [36]:
E = np.array([
    [9, 5, 2],
    [3, 8, 0],
    [0, 0, 0],
    [6, 7, -1],
])
E.shape

(4, 3)

a) Interpretar la matriz E,
explicando cómo han sido las ventas.  

- El cliente1 al mes compró 9 del artícilo1, 5 del artículo2 y 2 del artículo3
- El cliente2 al mes compró 3 del artícilo1, 8 del artículo2 y 0 del artículo3
- El cliente3 al mes compró 0 del artícilo1, 0 del artículo2 y 0 del artículo3
- El cliente4 al mes compró 6 del artícilo1, 7 del artículo2 y se tuvo una perdida equivalente al artículo3

***6) Una fábrica de coches produce tres modelos:
coupé, ranchera y económico. Cada coche
necesita las cantidades de cada uno de los
siguientes conceptos, relacionados en la matriz
C, en unidades convenientemente elegidas:
materiales, personal, impuestos y transporte.***

In [38]:
C = np.array([
    [7, 10, 5, 2],
    [8, 9, 3, 3],
    [5, 7, 2, 1]
])

V = np.array([
    [5],
    [15],
    [7],
    [2]
])

P = np.array([60,40,90]).reshape(1,3)


C.shape, P.shape, V.shape

((3, 4), (1, 3), (4, 1))

a) Las unidades semanales necesarias de cada
concepto. 


In [39]:
P.dot(C)

array([[1190, 1590,  600,  330]])

b) Los costes de un coche de cada modelo. 


In [40]:
C.dot(V)

array([[224],
       [202],
       [146]])

c) El coste total de la producción semanal.  

In [41]:
(P.dot(C)).dot(V)

array([[34660]])